kubernetes/cache/kube_cache.go

```diff
@@ -99,6 +99,7 @@ type KubeCache interface {
 
 	GetK8sGateway(namespace, name string) (*gatewayapi_v1.Gateway, error)
 	GetK8sGateways(namespace, labelSelector string) ([]*gatewayapi_v1.Gateway, error)
+	GetK8sGatewayClasses(labelSelector string) ([]*gatewayapi_v1.GatewayClass, error)
```

Comment:
This is the only function in the interface that gets resources without the context of a namespace. I've tried to implement this change without touching as much as possible, so I'm passing `""` into functions that have a namespace parameter, even if `clusterCacheLister` is `true`.

Reply:
If I understand the existing logic correctly, I think [this function that is used to get the cache](https://github.com/kiali/kiali/blob/5c9bed89e3f7e49864bf96d649076151bffa506c/kubernetes/cache/kube_cache.go#L620) uses the cluster-scoped cache when a namespace value of `""` is passed. Is my understanding correct?

I do think that this should use a cluster-scoped cache, as the resource is cluster-wide.